src/services/refactors/inlineVariable.ts

```diff
@@ -228,7 +231,13 @@ function getReplacementExpression(reference: Node, replacement: Expression): Exp
 
     // Functions also need to be parenthesized.
     // E.g.: const f = () => {}; f(); -> (() => {})();
-    if (isFunctionLike(replacement) && isCallLikeExpression(parent)) {
+    if (isFunctionLike(replacement) && (isCallLikeExpression(parent) || isPropertyAccessExpression(parent))) {
+        return factory.createParenthesizedExpression(replacement);
+    }
+
+    // Property access of literals need parentheses.
+    // E.g.: const x = 1; x.toString(); -> (1).toString();
+    if (isPropertyAccessExpression(parent) && (isLiteralTypeLikeExpression(replacement) || isObjectLiteralExpression(replacement))) {

```

Comment:
`isLiteralTypeLikeExpression` is not the right helper. Literal type-likes correspond to nodes that can be used in type position that are literals. You are more concerned with nodes in the expression space, and it just so happens that we use the same kinds of nodes in both places.

As far as I remember for property accesses, decimal numeric literals with no exponent or fractional component are really the only literals where you need to parenthesize.

```ts
"hello".toString(); // ✅
true.toString();    // ✅
false.toString();   // ✅
1.0.toString();     // ✅
1e32.toString();    // ✅
0xcafe.toString();  // ✅
```

But you can just say "only parenthesize when the node is a numeric literal" just to make this simpler.